code/modules/mob/living/carbon/human/human.dm

```diff
@@ -893,3 +893,10 @@
 			Weaken(10)
 		return 1
 	..()
+
+/mob/living/carbon/human/Bump(atom/A)
+	..()
+	var/obj/item/device/flightpack/FP = get_flightpack()
+	if(FP && FP.flight && FP.momentum_speed > 1)

```

Comment:
push those checks about speed and moment down into the flight_impact proc, they dont' belong here (they're internal state to the flight pack)

Side note, you can check the flying var here (or movement_type flag if you go the extra mile) to improve performance (i.e check if movement type is flying specifically, then you can get do this handling path)